src/test/java/com/iexec/sms/tee/config/TeeWorkerPipelineConfigurationTests.java

```diff
@@ -0,0 +1,165 @@
+    private static final String POST_FINGERPRINT = "postFingerprint";
+    private static final DataSize POST_HEAP_SIZE = DataSize.parse("3GB");
+    private static final String POST_ENTRYPOINT = "postEntrypoint";
+    private final TeeWorkerPipelineConfiguration.StageConfig preCompute = new TeeWorkerPipelineConfiguration.StageConfig(
+            PRE_IMAGE,
+            PRE_FINGERPRINT,
+            PRE_HEAP_SIZE,
+            PRE_ENTRYPOINT);
+    private final TeeWorkerPipelineConfiguration.StageConfig postCompute = new TeeWorkerPipelineConfiguration.StageConfig(
+            POST_IMAGE,
+            POST_FINGERPRINT,
+            POST_HEAP_SIZE,
+            POST_ENTRYPOINT);
+
+    private Validator validator;
+
+    @BeforeEach
+    void setUp() {
+        final ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    @Test
+    void shouldGetPipelineConfiguration() {
+        final TeeWorkerPipelineConfiguration.Pipeline pipeline = new TeeWorkerPipelineConfiguration.Pipeline(
+                VERSION,
+                preCompute,
+                postCompute
+        );
+        final TeeWorkerPipelineConfiguration config = new TeeWorkerPipelineConfiguration(Collections.singletonList(pipeline));
+
+        assertEquals(1, config.getPipelines().size());
+        final TeeWorkerPipelineConfiguration.Pipeline firstPipeline = config.getPipelines().get(0);
+        assertEquals(VERSION, firstPipeline.getVersion());
+
+        final TeeWorkerPipelineConfiguration.StageConfig preComputeConfig = firstPipeline.getPreCompute();
+        assertEquals(PRE_IMAGE, preComputeConfig.image());
+        assertEquals(PRE_FINGERPRINT, preComputeConfig.fingerprint());
+        assertEquals(PRE_HEAP_SIZE, preComputeConfig.heapSize());
+        assertEquals(PRE_ENTRYPOINT, preComputeConfig.entrypoint());
+
+        final TeeWorkerPipelineConfiguration.StageConfig postComputeConfig = firstPipeline.getPostCompute();
+        assertEquals(POST_IMAGE, postComputeConfig.image());
+        assertEquals(POST_FINGERPRINT, postComputeConfig.fingerprint());
+        assertEquals(POST_HEAP_SIZE, postComputeConfig.heapSize());
+        assertEquals(POST_ENTRYPOINT, postComputeConfig.entrypoint());
+    }
+
+    @Test
+    void shouldFailWhenPipelineConfigIsNull() {
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration>> violations = validator.validate(new TeeWorkerPipelineConfiguration(null));
+        assertFalse(violations.isEmpty());
+        assertThat(violations)
+                .extracting(ConstraintViolation::getMessage)
+                .containsExactly("Pipeline list must not be empty");
+    }
+
+    @Test
+    void shouldFailIfEmptyPipelineList() {
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration>> violations = validator.validate(new TeeWorkerPipelineConfiguration(Collections.emptyList()));
+        assertFalse(violations.isEmpty());
```

Comment:
Those statements (several occurences in the test class) may be more readable this way
```suggestion
        assertThat(violations).isNotEmpty();
```